apisix/plugins/proxy-rewrite.lua

```diff
@@ -278,15 +284,24 @@ function _M.rewrite(conf, ctx)
 
         local uri, _, err = re_sub(upstream_uri, conf.regex_uri[1],
                                    conf.regex_uri[2], "jo")
-        if uri then
-            upstream_uri = uri
-        else
+        if not uri then
             local msg = "failed to substitute the uri " .. ctx.var.uri ..
                         " (" .. conf.regex_uri[1] .. ") with " ..
                         conf.regex_uri[2] .. " : " .. err
             core.log.error(msg)
             return 500, {message = msg}
         end
+
+        local m, err = re_match(upstream_uri, conf.regex_uri[1], "jo")
+        if not m then
+            if err then
```

Comment:
We can merge the two conditions above?

Reply:
how? use the `re_match` first, then use the match result to replace?

Reply:
See https://github.com/apache/apisix/pull/9112#discussion_r1145538773